larray/core.py

```diff
@@ -4363,6 +4363,14 @@ def __getattr__(self, key):
         else:
             raise AttributeError("'{}' object has no attribute '{}'".format(self.__class__.__name__, key))
 
+    # needed to make *un*pickling work (because otherwise, __getattr__ is called before .axes exists, which leads to
+    # an infinite recursion)
+    def __getstate__(self):
+        return self.__dict__
+
+    def __setstate__(self, d):
+        self.__dict__ = d
+
```

Comment:
indirectly, yes. the Session pickling test includes Axis and LArray objects, which means they are pickled and unpickled too.